src/t8_geometry/t8_geometry_implementations/t8_geometry_mapping.hxx

```diff
@@ -0,0 +1,113 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_geometry_mapping.hxx
+ * TODO: Add description
+ */
+
+#ifndef T8_GEOMETRY_MAPPING_HXX
+#define T8_GEOMETRY_MAPPING_HXX
+
+#include <string>
+#include <vector>
+
+#include <t8.h>
+#include <t8_geometry/t8_geometry_with_vertices.hxx>
+#include <t8_geometry/t8_geometry_with_vertices.h>
+
+/* Abstract class for the mapping with Lagrange basis functions */
+class finite_element: public t8_geometry_with_vertices {
+ public:
+  /** 
+   * Constructor of the element geometry obtained by finite element mapping. The geometry
+   * is viable with all tree types and uses as many vertices as the tree type has.
+   * The vertices are saved via the \ref t8_cmesh_set_tree_vertices function.
+   * \param [in] dimension  0 <= \a dimension <= 3. Element dimension in the parametric space.
+   * \param [in] name       Element name.
+   */
+  finite_element (int dimension, const char *name);
+
+  /**
+   * Maps points in the reference space \f$ [0,1]^\mathrm{dim} \to \mathbb{R}^3 \f$.
+   * \param [in]  cmesh       The cmesh in which the point lies.
+   * \param [in]  gtreeid     The global tree (of the cmesh) in which the reference point is.
+   * \param [in]  ref_coords  Array of \a dimension x \a num_coords many entries, specifying points in \f$ [0,1]^\mathrm{dim} \f$.
+   * \param [in]  num_coords  Amount of points of /f$ \mathrm{dim} /f$ to map.
+   * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords. The length is \a num_coords * 3.
+   */
+  void
+  t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords, const size_t num_coords,
+                    double out_coords[3]) const;
```

Comment:
I don't remember where I saw it, but I remember there was an assertion somewhere to prevent the user from passing more than one point. That is why I gave it as [3]. I will change that parameter to a pointer, as you suggested.